src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +40,29 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
+    // Get logged-in user and other user
+    String loggedInUser = null;
+    String otherUser = request.getParameter("user");
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      loggedInUser = userService.getCurrentUser().toString();
```

Comment:
Based on the documentation here:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/users/User

It is unclear what `toString()` will return and whether there are any guarantees on the content/stability of the function. Based on the lack of promises/guarantees, it would seem that `toString()` should return something useful for debugging but may not be a stable API and may not return what you expect/want in the future.

I would recommend making this code more reliable and clear by explicitly retrieving a property that you care about such as `.getEmail()` or `.getUserId()`, depending on how you are identifying the user.